lib/langchain/llm/openai.rb

```diff
@@ -104,52 +114,14 @@ def complete(prompt:, **params)
     #
     # @param messages [Array<Hash>] List of messages comprising the conversation so far
     # @param model [String] ID of the model to use
-    def chat(
-      messages: [],
-      model: defaults[:chat_completion_model_name],
-      frequency_penalty: nil,
-      logit_bias: nil,
-      logprobs: nil,
-      top_logprobs: nil,
-      max_tokens: nil,
-      n: defaults[:n],
-      presence_penalty: nil,
-      response_format: nil,
-      seed: nil,
-      stop: nil,
-      stream: nil,
-      temperature: defaults[:temperature],
-      top_p: nil,
-      tools: [],
-      tool_choice: nil,
-      user: nil,
-      &block
-    )
-      raise ArgumentError.new("messages argument is required") if messages.empty?
-      raise ArgumentError.new("model argument is required") if model.empty?
-      raise ArgumentError.new("'tool_choice' is only allowed when 'tools' are specified.") if tool_choice && tools.empty?
+    def chat(params = {}, &block)
```

Comment:
I tried overriding the `model:` when calling the `chat()` method but it didn't work:
```ruby
irb(main):009> openai.chat messages:[{role:"user", content:"hello!"}], model: "gpt-4"
=>
#<Langchain::LLM::OpenAIResponse:0x0000000136817dd8
 @model=nil,
 @raw_response=
  {"id"=>"chatcmpl-9K2yrYv6ayNahExEymLN6ncBFSlqg",
   "object"=>"chat.completion",
   "created"=>1714565045,
   "model"=>"gpt-3.5-turbo-0125",
```

Reply:
Just pushed a fix.  Was overly caching calls to the UnifiedParameters#to_params method.